data_structures/src/chain.rs

```diff
@@ -2902,6 +2911,112 @@ impl ReputationEngine {
     pub fn clear_threshold_cache(&self) {
         self.threshold_cache.borrow_mut().clear_threshold_cache()
     }
+
+    /// Get ARS keys ordered by reputation. If tie, order by pkh.
+    pub fn get_rep_ordered_ars_list(&self) -> Vec<PublicKeyHash> {
+        self.ars
+            .active_identities()
+            .cloned()
+            .sorted_by_key(|&pkh| std::cmp::Reverse((self.trs.get(&pkh), pkh)))
+            .collect()
+    }
+}
+
+#[allow(clippy::cast_possible_truncation, clippy::cast_sign_loss)]
+/// This function returns the result of: y = mx + K
+/// The result is rounded and low saturated in 0
+fn magic_line(x: f64, m: f64, k: f64) -> u32 {
```

Comment:
I think in other places we are putting macros just between the documentation and the function signature.